### PR TITLE
Newsletters: add 253 (2023-05-31)

### PR DESCRIPTION
**Pull request by harding**
- [x] Lede, releases/RCs, topic links on Tuesday @harding 
- [x] Notable changes section ASAP @harding 
- [x] Waiting for confirmation: practical strategies when bidding for blockspace: fee bumping, size efficiency, fee-sensitive coin selection @Xekyo 
- [x] Bitcoin SE @bitschmidty

### CODE REVIEW

File: _posts/en/newsletters/2023-05-31-newsletter.md [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+      spender receives a refund.  For a receiver that doesn't trust both
+      the spender and the counterparty, they will need to wait for as
+      many confirmations as they would for a regular onchain
+      transaction.
+
+        Keceli and a commentator [link][keceli reply0] to
+        [previous][harding reply0] research describing how a zero-conf
+        payment can be made uneconomical to double spend by pairing it
+        with a fidelity bond that can be claimed by any miner who
+        observed both versions of the double spent transaction.  That
```
Comment (LarryRuane) [inline _posts/en/newsletters/2023-05-31-newsletter.md:57]:
```suggestion
        observed both versions of the double-spent transaction.  That
```

File: _posts/en/newsletters/2023-05-31-newsletter.md [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+      it's not possible to trustlessly receive a payment from the
+      joinpool (either offchain or onchain) without waiting for a
+      sufficient number of confirmations.  That can take hours, whereas
+      many LN payments currently complete in less than a second. Even
+      with fidelity bonds, LN would be faster on average.
+
+    - *Concern that the onchain footprint is high:* one [reply][jk_14]
+      noted that, at one transaction every five seconds, about 200 such
+      counterparties would consume the entire space of every block.
+      Another [reply][harding reply0] assumed the size of the
```
Comment (LarryRuane) [inline _posts/en/newsletters/2023-05-31-newsletter.md:114]:
```suggestion
      Another [reply][harding reply0] assumed the size of each of the
```

File: _posts/en/newsletters/2023-05-31-newsletter.md [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+      channel open or cooperative close transaction, so a counterparty
+      with a million users that creates 6.3 million onchain
+      transactions per year would use an equivalent amount of space to
+      each of those users opening or closing an average of 6.3 channels
+      each per year; thus, LN's onchain costs could be lower than using
+      the counterparty until it had reached massive scale.
+
+    - *Concern about a large hot wallet and the capital costs:* a
+      [reply][harding reply0] considered that the counterparty would
+      need to keep an amount of bitcoins on hand (probably in a hot
```
Comment (LarryRuane) [inline _posts/en/newsletters/2023-05-31-newsletter.md:125]:
```suggestion
      need to keep an amount of bitcoin on hand (probably in a hot
```

**Comment by Xekyo** [comment]
I have a draft for the series, will wait for Gloria to give it a read over and then add it here tomorrow morning.

File: _includes/specials/policy/en/03-bidding-for-block-space.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+prefer blockspace-efficient output types to reduce the future cost of
+spending their transaction outputs, but it’s during the [input
+selection][topic coin selection] that there is the most room to change
+the final composition and weight of the transaction. As transactions
+compete by feerate [fee/weight], a lighter transaction requires a lower
+fee to reach the same feerate.
+
+Some wallets, such as the Bitcoin Core wallet, try to combine inputs
+such that they avoid needing a change output altogether. Avoiding change
+saves the weight of an output now, but also saves the future cost of
```
Comment (glozow) [inline _includes/specials/policy/en/03-bidding-for-block-space.md:32]:
```suggestion
saves the weight of an output now, and saves the future cost of
```

File: _posts/en/newsletters/2023-05-31-newsletter.md [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+    space of managed joinpools.
+
+- **Transaction relay over Nostr:** Joost Jager [posted][jager nostr] to
+  the Bitcoin-Dev mailing list to request feedback on the idea by Ben
+  Carman of using the [Nostr][] protocol for relaying transactions that
+  might not propagate well on the P2P network of Bitcoin full nodes that
+  provide relay services.
+
+    In particular, Jager examines the possibility of using Nostr for the
+    relay of transaction packages---the ability to relay an ancestor
```
Comment (glozow) [inline _posts/en/newsletters/2023-05-31-newsletter.md:149]: As this is not the only use of relaying packages
```suggestion
    relay of transaction packages---in particular, the ability to relay an ancestor
```

File: _posts/en/newsletters/2023-05-31-newsletter.md [hunk 6]
```diff
@@ -237,26 +237,77 @@ Server][btcpay server repo], [BDK][bdk repo], [Bitcoin Improvement
...
+  This allows a peer who receives a compact block from one of its other
+  peers to request any missing transactions from us, which can help if
+  that other peer has become unresponsive.  This can help our peer
+  validate the block faster, which may also help them use it sooner in
+  time-critical functions, such as mining.
+
+- [Bitcoin Core #25796][] adds a new `descriptorprocesspsbt` that can be
+  used to update a [PSBT][topic psbt] with information that will help it
+  later be signed or finalized.  A [descriptor][topic descriptors]
+  provided to the RPC will be used to retrieve information from mempool
```
Comment (LarryRuane) [inline _posts/en/newsletters/2023-05-31-newsletter.md:260]:
```suggestion
  provided to the RPC will be used to retrieve information from the mempool
```

File: _includes/references.md [hunk 7]
```diff
@@ -44,6 +44,7 @@
 [libsecp256k1]: https://github.com/bitcoin-core/secp256k1
 [libsecp256k1 repo]: https://github.com/bitcoin-core/secp256k1
 [lnd repo]: https://github.com/lightningnetwork/lnd/
+[nostr]: https://github.com/nostr-protocol/nips
```
Comment (bitschmidty) [inline _includes/references.md:47]:
```suggestion
[nostr]: https://github.com/nostr-protocol/nostr
```

File: _posts/en/newsletters/2023-05-31-newsletter.md [hunk 8]
```diff
@@ -0,0 +1,328 @@
...
+  the Bitcoin-Dev mailing list to request feedback on the idea by Ben
+  Carman of using the [Nostr][] protocol for relaying transactions that
+  might not propagate well on the P2P network of Bitcoin full nodes that
+  provide relay services.
+
+    In particular, Jager examines the possibility of using Nostr for the
+    relay of transaction packages---the ability to relay an ancestor
+    transaction with a feerate below the minimum accepted value by
+    bundling it with a descendant that pays a high enough fee to
+    compensate for its relative's deficiency.  This makes [CPFP][topic
```
Comment (bitschmidty) [inline _posts/en/newsletters/2023-05-31-newsletter.md:152]: Maybe?

```suggestion
    compensate for its parent's deficiency.  This makes [CPFP][topic
```

File: _posts/en/newsletters/2023-05-31-newsletter.md [hunk 9]
```diff
@@ -237,26 +237,77 @@ Server][btcpay server repo], [BDK][bdk repo], [Bitcoin Improvement
...
+- [BTCPay Server 97e7e][] begins setting the [BIP78][] `minfeerate`
+  (minimum feerate) parameter for [payjoin][topic payjoin] payments.
+  See also the [bug report][btcpay server #4689] that lead to this
+  commit.
+
+- [BIPs #1446][] makes a small change and a number of additions to the
+  [BIP340][] specification of [schnorr signatures][topic schnorr
+  signatures] for Bitcoin-related protocols.  The change is allowing the
+  message to be signed to be an arbitrary length; previous versions of
+  the BIP required the message be exactly 32 bytes.  See a related
```
Comment (LarryRuane) [inline _posts/en/newsletters/2023-05-31-newsletter.md:296]:
```suggestion
  the BIP required the message to be exactly 32 bytes.  See a related
```
or
```suggestion
 the BIP required that the message be exactly 32 bytes.  See a related
```

File: _posts/en/newsletters/2023-05-31-newsletter.md [hunk 10]
```diff
@@ -237,26 +237,77 @@ Server][btcpay server repo], [BDK][bdk repo], [Bitcoin Improvement
...
+  change to the Libsecp256k1 library described in [Newsletter
+  #157][news157 libsecp].  The change has no effect on the use of BIP340
+  in consensus applications as signatures used with both [taproot][topic
+  taproot] and [tapscript][topic tapscript] (respectively, BIPs
+  [341][bip341] and [342][bip342]) use 32-byte messages.
+
+    The additions describe how to effectively use arbitrary length
+    messages, recommend how to use a hash tag prefix, and provide
```
Comment (LarryRuane) [inline _posts/en/newsletters/2023-05-31-newsletter.md:304]: not sure, but maybe
```suggestion
    messages, recommend how to use a hashtag prefix, and provide
```

File: _includes/specials/policy/en/03-bidding-for-block-space.md [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+  put links in this file
+  for any subheads use h3 (i.e., ###)
+  illustrations welcome (max width 800px)
+  if uncertain about anything, just do what seems best and harding will edit
+-->
+
+Last week we mentioned that transactions pay fees for the used
+blockspace rather than the transferred amount, and established that
+miners optimize their transaction selection to maximize collected fees.
+It follows that only those transactions get confirmed that reside in the
```
Comment (bitschmidty) [inline _includes/specials/policy/en/03-bidding-for-block-space.md:13]: maybe

It follows that only transactions that reside in the top of the mempool get confirmed when a block is found.

File: _includes/specials/policy/en/03-bidding-for-block-space.md [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+top of the mempool when a block is found. In this post, we will discuss
+practical strategies to get the most for our fees. Let’s assume we have
+a decent source of feerate estimates—we will talk more about feerate
+estimation in next week’s article.
+
+While constructing transactions, some parts of the transaction are more
+flexible than others. Every transaction requires the header fields, the
+recipient outputs are determined by the payments being made, and most
```
Comment (bitschmidty) [inline _includes/specials/policy/en/03-bidding-for-block-space.md:21]:
```suggestion
recipient outputs determined by the payments being made, and most
```

File: _includes/specials/policy/en/03-bidding-for-block-space.md [hunk 13]
```diff
@@ -0,0 +1,105 @@
...
+selection][topic coin selection] that there is the most room to change
+the final composition and weight of the transaction. As transactions
+compete by feerate [fee/weight], a lighter transaction requires a lower
+fee to reach the same feerate.
+
+Some wallets, such as the Bitcoin Core wallet, try to combine inputs
+such that they avoid needing a change output altogether. Avoiding change
+saves the weight of an output now, but also saves the future cost of
+spending the change output later. Unfortunately, such input combinations
+will only seldom be available unless the wallet sports a large UTXO pool
```
Comment (bitschmidty) [inline _includes/specials/policy/en/03-bidding-for-block-space.md:34]:
```suggestion
will seldom be available unless the wallet sports a large UTXO pool
```

File: _includes/specials/policy/en/03-bidding-for-block-space.md [hunk 14]
```diff
@@ -0,0 +1,105 @@
...
+with a broad variety of amounts.
+
+Modern output types are more blockspace-efficient than older output
+types. E.g. spending a P2TR input incurs less than 2/5ths of a P2PKH
+input’s weight. (Try it with our [transaction size
+calculator][]!) For multisig wallets, the recently
```
Comment (bitschmidty) [inline _includes/specials/policy/en/03-bidding-for-block-space.md:40]:
```suggestion
calculator][]) For multisig wallets, the recently
```

File: _includes/specials/policy/en/03-bidding-for-block-space.md [hunk 15]
```diff
@@ -0,0 +1,105 @@
...
+finalized [MuSig2][topic musig] schema and FROST protocol chalk out huge
```
Comment (bitschmidty) [inline _includes/specials/policy/en/03-bidding-for-block-space.md:41]:
```suggestion
finalized [MuSig2][topic musig] scheme and FROST protocol provide huge
```

File: _includes/specials/policy/en/03-bidding-for-block-space.md [hunk 16]
```diff
@@ -0,0 +1,105 @@
...
+achieve the lowest possible weight for the input set. Always selecting
+the lightest input set would locally minimize the cost of the current
+transaction, but also grind a wallet’s UTXO pool into small fragments.
+This could set the user up for transactions with huge input sets at high
+feerates later. Therefore, it is prescient for wallets to also select
+more and heavier inputs at low feerates to opportunistically consolidate
+funds into fewer modern outputs in anticipation of later blockspace
+demand spikes.
+
+High-volume wallets often batch multiple payments into a single
```
Comment (bitschmidty) [inline _includes/specials/policy/en/03-bidding-for-block-space.md:61]: Could link to https://bitcoinops.org/en/payment-batching/

File: _includes/specials/policy/en/03-bidding-for-block-space.md [hunk 17]
```diff
@@ -0,0 +1,105 @@
...
+transaction to reduce the transaction weight per payment. Instead of
+incurring the overhead of the header bytes and the change output for
+each payment, they only incur the overhead cost once shared across all
```
Comment (bitschmidty) [inline _includes/specials/policy/en/03-bidding-for-block-space.md:64]: maybe

```suggestion
each payment, they only incur the overhead cost once, shared across all
```

File: _includes/specials/policy/en/03-bidding-for-block-space.md (hunk 14 above)
Comment (Xekyo) [reply _includes/specials/policy/en/03-bidding-for-block-space.md:40]: That’s a whole sentence in the parentheses, so wouldn’t it need punctuation?

File: _includes/specials/policy/en/03-bidding-for-block-space.md (hunk 12 above)
Comment (glozow) [reply _includes/specials/policy/en/03-bidding-for-block-space.md:21]: I agree with @Xekyo

**Comment by bitschmidty** [comment]
Great newsletter, team! Thank to @harding @Xekyo @glozow @LarryRuane for your contributions!

**Pull request merged**